sys/luid/luid.c

```diff
@@ -44,19 +77,36 @@ void luid_custom(void *buf, size_t len, int gen)
     }
 }
 
-void luid_base(void *buf, size_t len)
+void luid_get_short(network_uint16_t *addr)
 {
-    assert(buf && (len > 0));
+    luid_base(addr, sizeof(*addr));
 
-    memset(buf, LUID_BACKUP_SEED, len);
+    /* https://tools.ietf.org/html/rfc4944#section-12 requires the first bit to
+     * 0 for unicast addresses */
+    addr->u8[0] &= 0x7F;
+    addr->u8[1] ^= lastused++;
+}
 
-#if CPUID_LEN
-    uint8_t *out = (uint8_t *)buf;
-    uint8_t cid[CPUID_LEN];
+void luid_get_eui48(eui48_t *addr)
+{
+    static uint8_t lastused;
+    if (!luid_get_eui48_custom(addr, lastused++)) {
+        luid_base(addr, sizeof(*addr));
+        addr->uint8[5] ^= lastused;
+     }
 
-    cpuid_get(cid);
-    for (size_t i = 0; i < CPUID_LEN; i++) {
-        out[i % len] ^= cid[i];
-    }
-#endif
+    eui48_set_local(addr);
+    eui48_clear_group(addr);
+}
+
+void luid_get_eui64(eui64_t *addr)
+{
+    static uint8_t lastused;
+    if (!luid_get_eui64_custom(addr, lastused++)) {
+        luid_base(addr, sizeof(*addr));
+        addr->uint8[7] ^= lastused;
+     }
+
+    addr->uint8[0] &= ~(0x01);
+    addr->uint8[0] |=  (0x02);
```

Comment:
Why not introduce a function that does that, like the ones in `eui64`. This would make the code more uniform.

Reply:
added.